chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -282,6 +289,35 @@ trait TransformerMacros extends MappingMacros with TargetConstructorMacros with
     }
   }
 
+  def expandValueClassToValueClass(
+      srcPrefixTree: Tree,
+      config: TransformerConfig
+  )(
+      From: Type,
+      To: Type
+  ): Either[Seq[TransformerDerivationError], Tree] = {
+
+    val fromValueClassMember = From.valueClassMember.toRight(
+      Seq(CantFindValueClassMember(From.typeSymbol.name.toString, To.typeSymbol.name.toString))
+    )
+
+    val toValueClassMember = To.valueClassMember.toRight(
+      Seq(CantFindValueClassMember(To.typeSymbol.name.toString, From.typeSymbol.name.toString))
+    )
+
+    for {
+      fromValueClassMember <- fromValueClassMember
+      fromValueClassMemberType = fromValueClassMember.returnType
+      toValueClassMemberType <- toValueClassMember.map(_.returnType)
+      fromMemberAccessTree = q"$srcPrefixTree.${fromValueClassMember.name}"
+      transformerBodyTree <- resolveRecursiveTransformerBody(fromMemberAccessTree, config)(
+        fromValueClassMemberType,
+        toValueClassMemberType
+      )
+      toCreationTree = q"new $To(${transformerBodyTree.tree})"
+    } yield mkTransformerBodyTree0(transformerBodyTree.target)(toCreationTree)
```

Comment:
This rule may require similar `mkTransformerBodyTree1` adjustment as in `expandTypeToValueClass` rule.
Please add at least at test that checks transformation between two value classes, relying on implicit partial transformer to transform their inner types.